src/www/ui/api/Controllers/CopyrightController.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/*
+ Author: Soham Banerjee <sohambanerjee4abc@hotmail.com>
+ SPDX-FileCopyrightText: © 2023 Soham Banerjee <sohambanerjee4abc@hotmail.com>
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+/**
+ * @file
+ * @brief Controller for copyright queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\Lib\Auth\Auth;
+use Fossology\Lib\Dao\UploadDao;
+use Fossology\Lib\Db\DbManager;
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+use Fossology\Lib\Data\Tree\ItemTreeBounds;
+
+
+class CopyrightController extends RestController
+{
+  /**
+   * @var ContainerInterface $container
+   * Slim container
+   */
+  protected $container;
+
+  /**
+   * @var copyrightHist $copyrightHist
+   * Copyright Histogram object
+   */
+  private $copyrightHist;
+
+  /**
+   * @var CopyrightDao $copyrightDao
+   * Copyright Dao object
+   */
+  private $copyrightDao;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->restHelper = $container->get('helper.restHelper');
+    $this->copyrightDao = $container->get('dao.copyright');
+    $this->copyrightHist = $this->restHelper->getPlugin('ajax-copyright-hist');
+  }
+
+  /**
+   * Update copyrights for a particular file
+   *
+   * @param  ServerRequestInterface $request
+   * @param  ResponseHelper         $response
+   * @param  array                  $args
+   * @return ResponseHelper
+   */
+
+  public function updateFileCopyrights($request, $response, $args)
+  {
+
+    $uploadTreeId = intval($args['itemId']);
+    $copyrightHash = $args['hash'];
+    $userId = $this->restHelper->getUserId();
+    $cpTable = $this->copyrightHist->getTableName('statement');
+    $returnVal = null;
+    $body = $this->getParsedBody($request);
+    $content = $body['content'];
+    try {
+      if (!$this->dbHelper->doesIdExist($this->restHelper->getUploadDao()->getuploadTreeTableName($uploadTreeId), "uploadtree_pk", $uploadTreeId)) {
+        $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+      }
```

Comment:
Please add a sanity check for upload ID as well, even though the backend doesn't need it.

Reply:
Sure I will add it